internal/backend/post.go

```diff
@@ -79,5 +85,36 @@ func (d *delphisBackend) NotifySubscribersOfCreatedPost(ctx context.Context, pos
 }
 
 func (d *delphisBackend) GetPostsByDiscussionID(ctx context.Context, discussionID string) ([]*model.Post, error) {
-	return d.db.GetPostsByDiscussionID(ctx, discussionID)
+	iter := d.db.GetPostsByDiscussionIDIter(ctx, discussionID)
+	return d.iterToPosts(ctx, iter)
+	//return d.db.GetPostsByDiscussionID(ctx, discussionID)
+}
+
+// Testing function to keep functionality
+func (d *delphisBackend) iterToPosts(ctx context.Context, iter datastore.PostIter) ([]*model.Post, error) {
+	var posts []*model.Post
+	post := model.Post{}
+
+	for iter.Next(&post) {
+		tempPost := post
+
+		// Check if there is a quotedPostID. Fetch if so
+		if tempPost.QuotedPostID != nil {
+			var err error
+			tempPost.QuotedPost, err = d.db.GetPostByID(ctx, *tempPost.QuotedPostID)
+			if err != nil {
+				// Do we want to fail the whole discussion if we can't get a quote?
+				return nil, err
+			}
+		}
+
+		posts = append(posts, &tempPost)
+	}
+
+	if err := iter.Close(); err != nil && err != io.EOF {
```

Comment:
OOC: Should we put `iter.Close()` as a defer somewhere to ensure it gets called? I could foresee memory leaks.